File: LeetCode/013.roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int size = s.size() - 1;
        int res = 0;
        int i;
        for (i = 0; i < size; i++){
            if (s[i] == 'I'){
                if (s[i + 1] == 'V')    res += 4, i++;
                else if (s[i + 1] == 'X')    res += 9, i++;
                else res += 1;
            }else if (s[i] == 'V'){
                res += 5;
            }else if (s[i] == 'X'){
                if (s[i + 1] == 'L')    res += 40, i++;
                else if (s[i + 1] == 'C')   res += 90, i++;
                else res += 10;
            }else if (s[i] == 'L'){
                res += 50;
            }else if (s[i] == 'C'){
                if (s[i + 1] == 'D')    res += 400, i++;
                else if (s[i + 1] == 'M')   res += 900, i++;
                else res += 100;
            }else if (s[i] == 'D'){
                res += 500;
            }else if (s[i] == 'M'){
                res += 1000;
            }
        }
    
        if (i <= size){
            if (s[size] == 'I') res += 1;
            else if (s[size] == 'V')    res += 5;
            else if (s[size] == 'X')    res += 10;
            else if (s[size] == 'L')    res += 50; 
            else if (s[size] == 'C')    res += 100;
            else if (s[size] == 'D')    res += 500;
            else if (s[size] == 'M')    res += 1000;
        }
        

        return res;

    }
};

File: LeetCode/0013.roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int length = s.length() - 1;
        char[] cstr = s.toCharArray();
        int res = 0;
        int i;
        for (i = 0; i < length; i++){
            if (cstr[i] == 'I'){
                if (cstr[i + 1] == 'V') {
                    res += 4; i++;
                }
                else if (cstr[i + 1] == 'X'){
                    res += 9; i++;
                }    
                else res += 1;
            }else if (cstr[i] == 'V'){
                res += 5;
            }else if (cstr[i] == 'X'){
                if (cstr[i + 1] == 'L') {
                    res += 40; i++;
                }
                else if (cstr[i + 1] == 'C'){
                    res += 90; i++;
                }
                else res += 10;
            }else if (cstr[i] == 'L'){
                res += 50;
            }else if (cstr[i] == 'C'){
                if (cstr[i + 1] == 'D') {
                    res += 400; i++;
                }
                else if (cstr[i + 1] == 'M') {
                    res += 900; i++;
                }
                else res += 100;
            }else if (cstr[i] == 'D'){
                res += 500;
            }else if (cstr[i] == 'M'){
                res += 1000;
            }
        }

        if (i <= length){
            if (cstr[length] == 'I')    res += 1;
            else if (cstr[length] == 'V')   res += 5;
            else if (cstr[length] == 'X')   res += 10;
            else if (cstr[length] == 'L')   res += 50;
            else if (cstr[length] == 'C')   res += 100;
            else if (cstr[length] == 'D')   res += 500;
            else if (cstr[length] == 'M')   res += 1000;
        }

        return res;
    }
}